dataRetriever/resolvers/validatorInfoResolver.go

```diff
@@ -121,7 +150,81 @@ func (res *validatorInfoResolver) resolveHashRequest(hash []byte, epoch uint32,
 		return err
 	}
 
-	return res.marshalAndSend(data, pid)
+	return res.marshalAndSend([][]byte{data}, pid)
+}
+
+// resolveMultipleHashesRequest sends the response for a hash array type request
+func (res *validatorInfoResolver) resolveMultipleHashesRequest(hashesBuff []byte, epoch uint32, pid core.PeerID) error {
+	b := batch.Batch{}
+	err := res.marshalizer.Unmarshal(&b, hashesBuff)
+	if err != nil {
+		return err
+	}
+	hashes := b.Data
+
+	validatorInfoForHashes, err := res.fetchValidatorInfoForHashes(hashes, epoch)
+	if err != nil {
+		return fmt.Errorf("resolveMultipleHashesRequest error %w from buff %s", err, hashesBuff)
+	}
+
+	return res.sendValidatorInfoForHashes(validatorInfoForHashes, pid)
+}
+
+func (res *validatorInfoResolver) sendValidatorInfoForHashes(validatorInfoForHashes [][]byte, pid core.PeerID) error {
+	if len(validatorInfoForHashes) > res.maxNumOfValidatorInfoInResponse {
```

Comment:
instead of using this fixed size variant, why not use the dataPacker implementation as seen in the transactionResolver.go. The maxNumOfValidatorInfoInResponse + config will no longer be necessary as you can define a maximum package size const.